src/test/java/reposense/model/RepoLocationTest.java

```diff
@@ -1,49 +1,149 @@
 package reposense.model;
 
+import static reposense.model.RepoLocation.isLocalRepo;
+
 import org.junit.Assert;
+import org.junit.Assume;
 import org.junit.Test;
 
 import reposense.parser.InvalidLocationException;
 import reposense.util.AssertUtil;
+import reposense.util.SystemUtil;
 
 public class RepoLocationTest {
+
+    private static final String LOCAL_REPO_LOCATION_VALID_WITHOUT_DOT_GIT_ONE = "repo";
+    private static final String LOCAL_REPO_LOCATION_VALID_WITHOUT_DOT_GIT_TWO = "../path/to/repo";
+    private static final String LOCAL_REPO_LOCATION_VALID_WITHOUT_DOT_GIT_THREE = "file://path/to/repo";
+    private static final String LOCAL_REPO_LOCATION_VALID_WITH_DOT_GIT_ONE = "path/to/repo/.git";
+    private static final String LOCAL_REPO_LOCATION_VALID_WITH_DOT_GIT_TWO = "file://path/to/repo.git";
```

Comment:
I don't think this is a valid path. Using [RFC 8089](https://datatracker.ietf.org/doc/html/rfc8089) under Appendix B, there should be 3 slashes after `file:` (i.e. `file:///path/to/repo.git`). If there are only 2, then the hostname immediately follows (i.e. `file://localhost/path/to/repo.git`). This is also the same for Windows.

Is it also possible to add a test that uses a relative path? Something like `file:///path/another/../to/repo.git` or `https://github.com/reposense/wrong/path/../../RepoSense.git`. I'm not sure how the parsing will be affected, but it should still have the same expected repo and organization name.

Reply:
> I don't think this is a valid path.

Admittedly, I'm not too well-versed in this.
![image](https://user-images.githubusercontent.com/65345505/153747441-a7ba3937-da06-486d-9740-39f448120529.png) but these two both seem to work and that was why I tested it as such. My interpretation of it was that `/path/to/repo/` was a generic path and `file://` was prepended to it. 

However, 1 slash don't seem to work as shown below:
![image](https://user-images.githubusercontent.com/65345505/153747683-bf302a3d-c9cf-48b2-805f-83c6e23dfaa1.png)

even though they should as seen from page 17 of the same [document](https://datatracker.ietf.org/doc/html/rfc8089#page-18), which seems to suggest that the single slash is valid. So maybe git clone does not strictly follow the convention?

Reply:
> Is it also possible to add a test that uses a relative path?

Noted on this. I did not think about such a scenario. Just to confirm, the goal is to ensure that whatever output folder is generated lines up with the generated name so there are no name mismatch after analysis. Because this `https://github.com/reposense/wrong/path/../../RepoSense.git` will produce an 'organization' like `reposense-wrong-path-..-..` even though it's basically the same as `reposense`. 

I'm not too sure if manually collapsing it would be easy. Would like to clarify which situation you are referring to.

Reply:
> So maybe git clone does not strictly follow the convention?

I think probably we might be spending too much effort into a feature that might not be too commonly used. Probably let's just make sure `file://` and `file:///` works, and let GitClone throw an exception if any edge cases were to appear.

> It seems one way to do it might be to wrap the given `filepath` for organization into a Path object, use `normalize()`

Yup, I think this would be ideal? It's probably very rare someone does this but since we have `normalize()` I think it should be worth to use it.

Reply:
> Probably let's just make sure `file://` and `file:///` works, and let GitClone throw an exception if any edge cases were to appear.

I agree. I think it is sufficient to provide a working case for the above scenarios and for everything else, a 'good enough' guess would be sufficient if it the location provided actually works with git clone but is not one of the cases we specified.  I have chosen to use `file:/*` instead which always strips all `/` in front of `file:` and treats everything after as the path to make handling easier since none of this actually changes the argument used for the actual `git clone` that we do. 

> Yup, I think this would be ideal? It's probably very rare someone does this but since we have normalize() I think it should be worth to use it.

I have implemented it in the update and added test cases for it.

@dcshzj I have updated the PR and would like another review. Thanks!